tests/ecmp/test_fgnhg.py

```diff
@@ -318,25 +314,97 @@ def common_setup_teardown(tbinfo, duthosts, rand_one_dut_hostname):
+def test_fg_ecmp(tbinfo, common_setup_teardown, ptfadapter, ptfhost):
     duthost, cfg_facts, router_mac, net_ports = common_setup_teardown
 
-    # IPv4 test
-    port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts, router_mac, net_ports, DEFAULT_VLAN_IPv4, PREFIX_IPv4)
-    fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv4) 
+    if MINIGRAPH_PARSING == "OFF":
+        # IPv4 test
+        port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts,
+                                                                            router_mac, net_ports, DEFAULT_VLAN_IPv4,
+                                                                            PREFIX_IPv4)
+        fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv4)
+
+        # IPv6 test
+        port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts,
+                                                                            router_mac, net_ports, DEFAULT_VLAN_IPv6,
+                                                                            PREFIX_IPv6)
+        fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv6)
+
+    else:
 
-    # IPv6 test
-    port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts, router_mac, net_ports, DEFAULT_VLAN_IPv6, PREFIX_IPv6)
-    fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv6) 
+        mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+    
+        prefix_ipv4 = ""
+        prefix_ipv6 = ""
+
+        try:
+            nhg_prefixes = mg_facts['FG_NHG_PREFIX']
+        except Exception as e:
+            logger.info("Minigraph does not contain the required ECMP entries. Terminating test...")
+            pytest.skip("")
+
+        for nhg_prefix in nhg_prefixes.keys():
+            if "." in nhg_prefix:
+                prefix_ipv4 = nhg_prefix
+            else:
+                prefix_ipv6 = nhg_prefix
+
+        bank_0_port_ipv4 = []
+        bank_1_port_ipv4 = []
+        bank_0_port_ipv6 = []
+        bank_1_port_ipv6 = []
+        ipv4_to_port = {}
+        ipv6_to_port = {}
+
+        port_list = mg_facts['PORT_NHIPV4'].keys()
+        port_list = [int(k.replace('etp', '')) for k in port_list]
+        nhipv4_port = dict((v, k) for k, v in mg_facts['PORT_NHIPV4'].items())
+        nhipv6_port = dict((v, k) for k, v in mg_facts['PORT_NHIPV6'].items())
+        for ip, bank_mapping in mg_facts['FG_NHG_MEMBER'].iteritems():
+            if ip in nhipv4_port and bank_mapping['bank'] == 0:
+                bank_0_port_ipv4.append(int(nhipv4_port[ip].replace('etp', '')))
```

Comment:
What is the output of the etp command here, I have some doubts about how this is functioning. bank_x_port lists the ptf interface id, etpxx does not correspond to the ptf interface id and they in fact differ by 1. Eg etp1 corresponds to interface 0 of ptf. 
Why do we not get Ethernetxx interface names?
Can we refer to the configure_interfaces to use similar logic to obtain the interface name?

Reply:
cfg_facts['port_index_map']  offers a Ethernet to etp map but not an etp to ptf port map.
Is there some other way of getting this mapping? We may need to add some checks for port range
that is entered in the parsed minigraph itself

Reply:
cfg_facts['port_index_map'] offers a map from Ethernet to index, starting from 0. This corresponds to the ptf interfaces and does not offer a "Ethernet to etp map". REF: https://github.com/Azure/sonic-mgmt/blob/ed0d49321a25d21e9e331c523863b28c7e2c0789/ansible/library/config_facts.py#L75 

On whether there is an etp to Ethernet to ptf index map, I am not sure on that. I would recommend checking some of the other tests if they have something like this, I will try to check it as well. One suggestion I have is that the "link" field in FG_NHG_MEMBER in cfg_facts may already contain what you need here since it will have an Ethernetxx mapped to the IP.